src/web/components/KeyPairs/KeyPairDialog.tsx

```diff
@@ -1,47 +1,62 @@
-import { useState } from 'react';
-import { SubmitHandler } from 'react-hook-form';
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
 
 import { AddKeyPairFormProps } from '../../services/keyPairService';
 import { Dialog } from '../Core/Dialog';
-import { Form } from '../Core/Form';
 import { TextInput } from '../Input/TextInput';
 import { KeyPairModel } from './KeyPairModel';
 
 import './KeyPairDialog.scss';
 
-type AddKeyPairDialogProps = {
+type AddKeyPairDialogProps = Readonly<{
   onAddKeyPair: (form: AddKeyPairFormProps) => Promise<void>;
   triggerButton: JSX.Element;
   keyPair?: KeyPairModel;
-};
+}>;
 
-type KeyPairDialogProps = AddKeyPairDialogProps;
-
-function KeyPairDialog(props: KeyPairDialogProps) {
+function KeyPairDialog({ onAddKeyPair, triggerButton, keyPair }: AddKeyPairDialogProps) {
   const [open, setOpen] = useState(false);
-  const { keyPair, onAddKeyPair } = props;
 
-  const onSubmit: SubmitHandler<AddKeyPairFormProps> = async (formData) => {
+  const formMethods = useForm<AddKeyPairFormProps>({
+    defaultValues: { name: keyPair?.name },
+  });
+  const { handleSubmit, setValue, reset } = formMethods;
+
+  useEffect(() => {
```

Comment:
I'm guessing this is to work around state still being around if you close the dialog and open it again? If so, I'd prefer a better solution - this risks us still having state floating around if we add more controls and forget to clear them here.

I think the real problem is that we keep the dialog around even after it's closed. It's not a great pattern and this is a good example why.

If you could switch to not rendering the dialog in whatever uses it, and controlling state from there, the problem would go away for good.

It'd be the classic {showKeyPairDialog && <KeyPairDialog ... />} instead of the current system, but you'd need to provide the button and move the showKeyPairDialog state to the parent. Give it a go but if it turns into too much of a change, just leave it like this and we can fix it another time.

Reply:
Changed the structure for this ticket.  Tech debt ticket exists to change that pattern in the other dialogs